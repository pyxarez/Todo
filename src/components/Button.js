import React from 'react';

import './Button.res/style.css'

Button.propTypes = {
  type: React.PropTypes.string.isRequired,
  onClick: React.PropTypes.func,
}

export default function Button({type, onClick}) {
  return  (
    <button
      className={`my-button-component button-${type}`}
      onClick={onClick}/>
  );
}
