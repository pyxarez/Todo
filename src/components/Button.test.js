import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

const setup = () => {
  const props = {
    onClick: () => {},
    type: 'swap'
  };

  const enzymeWrapper = shallow(<Button {...props}/>)

  return {
    props,
    enzymeWrapper
  }
};

describe('Component :: Button', () => {
  it('should render self', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});