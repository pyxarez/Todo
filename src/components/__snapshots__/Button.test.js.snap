exports[`Component :: Button should render self 1`] = `
<button
  className="my-button-component button-swap"
  onClick={[Function]} />
`;
